src/playground-ide.ts

```diff
@@ -153,9 +153,60 @@ export class PlaygroundIde extends LitElement {
 
   /**
    * A document-relative path to a project configuration file.
+   *
+   * When both `projectSrc` and `files` are set, the one set most recently wins.
+   * Slotted children win only if both `projectSrc` and `files` are undefined.
    */
-  @property({attribute: 'project-src'})
-  projectSrc?: string;
+  @property({attribute: 'project-src', hasChanged: () => false})
+  get projectSrc(): string | undefined {
+    // To minimize synchronization complexity, we delegate the `projectSrc` and
+    // `files` getters/setters directly to our <playground-project>. The only
+    // case we need to handle is properties set before upgrade or before we
+    // first render the <playground-project>.
+    //
+    // TODO(aomarks) Maybe a "delegate" decorator for this pattern?
+    const project = this._project;
+    if (project) {
+      return project.projectSrc;
+    } else {
+      // We haven't rendered yet.
+      return this._projectSrcSetBeforeRender;
+    }
+  }
+
+  set projectSrc(src: string | undefined) {
+    const project = this._project;
+    if (project) {
```

Comment:
can `project` ever go from being defined to undefined?

Reply:
No